syntax = "proto3";

package example;

import "google/protobuf/empty.proto";

service Source {
}


service SourceOld {
    /*
    returns the time when the next order is in queue or none if no orders availale or a vehicle is on route
    */
    rpc next_order(google.protobuf.Empty) returns (Time);

    /*
    loads the next part to the carrier, resets the on_route flag
    */
    rpc get_part(google.protobuf.Empty) returns (Data);

    /*
    sets the on_route flag on parent
    */
    rpc reserve_next(google.protobuf.Empty) returns (Data);
}


service Sink {
    /*
    loads coordinates of sink
    */
    rpc get_coordinates(google.protobuf.Empty) returns (Data);

    /*
    puts a part into the sink. if it is full it returns a time to wait
    */
    rpc put_part(google.protobuf.Empty) returns (Time);
}


service Vehicle {
    /*
    loads coordinates of vehicle, if vehicle is idle, otherwise returns a not available error (closes vehicle first)
    */
    rpc get_eta(Data) returns (Time);

    /*
    drive to requested position and pickup a workpiece
    */
    rpc drive(Data) returns (google.protobuf.Empty);
}


message Data {
    string name = 1;
    float x = 2;
    float y = 3;
}


message Time {
    float time = 1;
}
